just_audio/lib/just_audio.dart

```diff
@@ -512,12 +512,13 @@ class AudioPlayer {
         controller.close();
         return;
       }
-      controller.add(position);
+      if (playing) {
+        controller.add(position);
+      }
```

Comment:
I understand the idea of this, but what if the user seeks while not playing? Currently the stream over-eagerly emits position events, and then `.distinct()` is used to cancel them out. So that guarantees the correct position updates, but is not the most efficient. So you're onto something here in terms of efficiency, but we will need some test cases for paused position changes based on seeking within a track and skipping between tracks.

Reply:
Yeah, I checked that and saw that user handling is still taken into account by the `playbackEventStream`. However, I didn't exactly know that `.distinct()` filters out redundant events. I'll add some tests then.